Packages/com.unity.inputsystem/InputSystem/Editor/ProjectWideActions/ProjectWideActionsAsset.cs

```diff
@@ -39,14 +34,39 @@ static void OnPostprocessAllAssets(string[] importedAssets, string[] deletedAsse
 
             internal static void MoveInputManagerAssetActionsToProjectWideInputActionAsset()
             {
-                var objects = AssetDatabase.LoadAllAssetsAtPath(kAssetPath);
+                var objects = AssetDatabase.LoadAllAssetsAtPath(kAssetPathInputManager);
                 if (objects != null)
                 {
                     var inputActionsAsset = objects.FirstOrDefault(o => o != null && o.name == kAssetName) as InputActionAsset;
-                    if (inputActionsAsset != null)
+                    if (inputActionsAsset != default)
                     {
-                        var json = JsonUtility.ToJson(inputActionsAsset, prettyPrint: true);
-                        File.WriteAllText(ProjectWideActionsAsset.kDefaultAssetPath, json);
+                        // Found some actions in the InputManager.asset file
+                        //
+                        string path = ProjectWideActionsAsset.kDefaultAssetPath;
+
+                        if (File.Exists(path))
+                        {
+                            // We already have a path containing inputactions, find a new unique filename
+                            //
+                            //  eg  Assets/InputSystem_Actions.inputactions ->
+                            //      Assets/InputSystem_Actions (1).inputactions ->
+                            //      Assets/InputSystem_Actions (2).inputactions ...
+                            //
+                            string[] files = Directory.GetFiles("Assets", "*.inputactions");
+                            List<string> names = new List<string>();
+                            for (int i = 0; i < files.Length; i++)
+                            {
+                                names.Add(System.IO.Path.GetFileNameWithoutExtension(files[i]));
+                            }
+                            string unique = ObjectNames.GetUniqueName(names.ToArray(), "InputSystem_Actions");
+                            path = "Assets/" + unique + ".inputactions";
+                        }
+
+                        var json = inputActionsAsset.ToJson();
+                        File.WriteAllText(path, json);
+
+                        AssetDatabase.Refresh();
+                        InputSystem.actions = (InputActionAsset)AssetDatabase.LoadAssetAtPath(path, typeof(InputActionAsset));
```

Comment:
I wonder if we shouldn't provide a Debug.Info message here informing about the conversion that have been done and that the asset has been assigned to InputSystem.actions?

Also, what if InputSystem.actions is already assigned? In that case I do not think we should assign it, only port to a new asset and inform about the conversion that has been made.

Reply:
Happy to add a message, but I can't find any example of "Debug.Info" - can you provide an example?

As for the assignment, we are in this block of code because we found project wide settings in the input manager asset file.

When testing this with @lyndon-unity we worked on the assumption that the user probably did want the assignment to be updated.

Reply:
Thinking about this, maybe just let it be as is

Reply:
I've added a Debug.Log() for both of these: migration (and assignment if required)